### ISSUE TITLE
Ubuntu 22.04 PETSC 3.19.6 compiler warnings

### ISSUE THREAD

**Issue by fcooper8472** [Chaste/Chaste#263]
Warnings are now emitted since PETSC 3.19:

```
[ 66%] Building CXX object global/CMakeFiles/chaste_global.dir/src/fortests/PetscSetupUtils.cpp.o
/home/fcooper/GitRepos/Chaste/Chaste/global/src/fortests/PetscSetupUtils.cpp:80:13: warning: PETSC_NULL is deprecated, use PETSC_NULLPTR instead (since version 3.19)
   80 |     PETSCEXCEPT(PetscInitialize(p_args->p_argc, p_args->p_argv, PETSC_NULL, PETSC_NULL));
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This appears to have only been [introduced in 3.17](https://petsc.org/main/changes/317/), so we will need to (for instance) make it a PETSC version-specific macro in a new header file.

**Comment by fcooper8472** [comment]
Something along these lines:

```cpp
// Use PETSC_NULLPTR for PETSc 3.19 and later, and PETSC_NULL for older versions
#if PETSC_VERSION_GE(3, 19, 0)
    #define CHASTE_PETSC_NULLPTR PETSC_NULLPTR
#else
    #define CHASTE_PETSC_NULLPTR PETSC_NULL
#endif

```

**Comment by fcooper8472** [comment]
Closed by #276
